lib/ex_unit/lib/ex_unit/capture_server.ex

```diff
@@ -231,6 +231,8 @@ defmodule ExUnit.CaptureServer do
   ## :logger handler callback.
 
   def log(event, _config) do
+    trap_exits = Process.flag(:trap_exit, false)
```

Comment:
I don't think we should toggle this because if a linked process exits, then it will make this process crash, while the process wants to trap exits.

I believe the best answer here is to store the tasks before calling `Task.await_many(:infinity)` and then do a traversal that discards the trapped message if trapping exits.